### DISCUSSION TITLE
Contribution Guide

### DISCUSSION THREAD

**Discussion by lufinkey** [lostb1t/replex#147] [General]
Can you make a guide for contributing features to this project? I was wanting to hook into a few API endpoints and add custom data, but I'm just having a bit of trouble figuring out the project and where I would put those changes. If there was a guide that outlined the basic structure of the code it would be really helpful!

**Comment by lostb1t** [comment]
if you can let me know what your plan is then i can walk you through the steps.

**Comment by lufinkey** [comment]
So the main feature I was trying to add was to make a "pseudo-library" that showed content from all a user's servers. But I know it would probably mean you would need the user's plex web api token to fetch their servers, so it would be kind of a big change

but for a guide the main things I would be looking for are
- where to add hooks into specific requests or data collections. I saw the transform methods and the routes, but wasn't sure whether I should be hooking into a wildcard route or if I should use an existing handler for a specific type (hubs, etc) to put those things in
- how I should interact with the cache and structure things for it, or do caching in general

I might just make my own just for fun and point it at the replex setup to keep all the features though 💯
